src/EditorFeatures/CSharpTest/MakeStructReadOnly/MakeStructReadOnlyTests.cs

```diff
@@ -0,0 +1,691 @@
+{
+}
+");
+    }
+
+    [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructReadOnly)]
+    public async Task TestOnStructWithReadOnlyField()
+    {
+        await TestAsync(
+@"struct [|S|]
+{
+    readonly int i;
+}",
+@"readonly struct S
+{
+    readonly int i;
+}");
+    }
+
+    [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructReadOnly)]
+    public async Task TestOnRecordStructWithReadOnlyField()
+    {
+        await TestAsync(
+@"record struct [|S|]
+{
+    readonly int i;
+}",
+@"readonly record struct S
+{
+    readonly int i;
+}");
+    }
+
+    [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructReadOnly)]
+    public async Task TestOnStructWithGetOnlyProperty()
+    {
+        await TestAsync(
+@"struct [|S|]
+{
+    int P { get; }
+}",
+@"readonly struct S
+{
+    int P { get; }
+}");
+    }
+
+    [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructReadOnly)]
+    public async Task TestOnRecordStructWithGetOnlyProperty()
+    {
+        await TestAsync(
+@"record struct [|S|]
+{
+    int P { get; }
+}",
+@"readonly record struct S
+{
+    int P { get; }
+}");
+    }
+
+    [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructReadOnly)]
+    public async Task TestOnStructWithInitOnlyProperty()
+    {
+        await TestAsync(
+@"struct [|S|]
+{
+    int P { get; init; }
+}
+
+namespace System.Runtime.CompilerServices
+{
+    public sealed class IsExternalInit
+    {
+    }
+}",
+@"readonly struct S
+{
+    int P { get; init; }
+}
+
+namespace System.Runtime.CompilerServices
+{
+    public sealed class IsExternalInit
+    {
+    }
+}");
+    }
+
+    [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructReadOnly)]
+    public async Task TestOnRecordStructWithInitOnlyProperty()
+    {
+        await TestAsync(
+@"record struct [|S|]
+{
+    int P { get; init; }
+}
+
+namespace System.Runtime.CompilerServices
+{
+    public sealed class IsExternalInit
+    {
+    }
+}",
+@"readonly record struct S
+{
+    int P { get; init; }
+}
+
+namespace System.Runtime.CompilerServices
+{
+    public sealed class IsExternalInit
+    {
+    }
+}");
+    }
+
+    [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructReadOnly)]
+    public async Task TestOnRecordStructWithReadOnlyField2()
+    {
+        await TestAsync(
+@"record struct [|S|]
+{
+    readonly int i;
+}",
+@"readonly record struct S
+{
+    readonly int i;
+}");
+    }
+
+    [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructReadOnly)]
+    public async Task TestMissingRecordStructWithPrimaryConstructorField()
+    {
+        await TestMissingAsync(
+@"record struct S(int i)
+{
+}");
+    }
+
+    [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMakeStructReadOnly)]
+    public async Task TestMissingOnRecordStructWithPrimaryConstructorFieldAndNormalField()
+    {
+        await TestMissingAsync(
+@"record struct S(int i)
+{
+    readonly int j;
+}");
```

Comment:
Why this can't be readonly?

Reply:
because `int i` becomes a writable member in a record-struct.